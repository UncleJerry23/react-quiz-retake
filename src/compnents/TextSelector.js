import React from 'react';
import PropTypes from 'prop-types';

function TextSelector({ text }) {
  return <p>{text}</p>;
}

TextSelector.propTypes = {
  text: PropTypes.string.isRequired
};

export default TextSelector;
