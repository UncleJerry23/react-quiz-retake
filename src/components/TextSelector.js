import React from 'react';
import PropTypes from 'prop-types';

function TextSelector({ state, handleChange }) {
  return (
    <>
      <input type="text" name="headerText" value={state.headerText} placeholder="Header Text" onChange={handleChange}/>
      <input type="text" name="footerText" value={state.footerText} placeholder="Footer Text" onChange={handleChange}/>
      <input type="text" name="image" value={state.image} placeholder="Image Url" onChange={handleChange}/>
    </>
  );
}

TextSelector.propTypes = {
  state: PropTypes.shape({
    headerText: PropTypes.string.isRequired,
    footerText: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired
  }).isRequired,
  handleChange: PropTypes.func.isRequired
};

export default TextSelector;
