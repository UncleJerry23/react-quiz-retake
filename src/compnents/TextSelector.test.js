import React from 'react';
import { shallow } from 'enzyme';
import TextSelector from './TextSelector';

describe('TextSelector component', () => {
  it('renders a TextSelector', () => {
    const text = 'this is text i promise';

    const wrapper = shallow(<TextSelector text={text} />);
    expect(wrapper).toMatchSnapshot();
  });
});
