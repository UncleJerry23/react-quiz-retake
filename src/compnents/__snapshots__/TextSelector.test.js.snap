// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextSelector component renders a TextSelector 1`] = `
<p>
  this is text i promise
</p>
`;
